src/main/java/org/voltdb/exportclient/KinesisFirehoseExportClient.java

```diff
@@ -0,0 +1,264 @@
+
+public class KinesisFirehoseExportClient extends ExportClientBase {
+    private static final VoltLogger m_logger = new VoltLogger("ExportClient");
+
+    String m_streamName = null;
+    AmazonKinesisFirehoseClient m_firehoseClient = null;
+    boolean m_skipInternals;
+    ExportDecoderBase.BinaryEncoding m_binaryEncoding;
+    char m_seperator;
+    // use thread-local to avoid SimpleDateFormat thread-safety issues
+    ThreadLocal<SimpleDateFormat> m_ODBCDateformat;
+
+    @Override
+    public void configure(Properties config) throws Exception
+    {
+        String regionName = config.getProperty("region","").trim();
+        if (regionName.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide an region");
+        }
+        Region region = RegionUtils.getRegion(regionName);
+
+        m_streamName = config.getProperty("stream.name","").trim();
+        if (m_streamName.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide a stream.name");
+        }
+
+        String accessKey = config.getProperty("access.key","").trim();
+        if (accessKey.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide an access.key");
+        }
+        String secretKey = config.getProperty("secret.key","").trim();
+        if (secretKey.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide a secret.key");
+        }
+        m_firehoseClient = new AmazonKinesisFirehoseClient(
+                new BasicAWSCredentials(accessKey, secretKey));
+        m_firehoseClient.setRegion(region);
+
+        m_skipInternals = Boolean.parseBoolean(config.getProperty("skipinternals", "false"));
+
+        final TimeZone tz = TimeZone.getTimeZone(config.getProperty("timezone", VoltDB.GMT_TIMEZONE.getID()));
+        m_ODBCDateformat = new ThreadLocal<SimpleDateFormat>() {
+            @Override
+            protected SimpleDateFormat initialValue() {
+                SimpleDateFormat sdf = new SimpleDateFormat(Constants.ODBC_DATE_FORMAT_STRING);
+                sdf.setTimeZone(tz);
+                return sdf;
+            }
+        };
+
+        m_binaryEncoding = ExportDecoderBase.BinaryEncoding.valueOf(
+                config.getProperty("binaryencoding", "HEX").trim().toUpperCase());
+
+        // Default to CSV if missing
+        String type = config.getProperty("type", "csv").trim();
+        if (type.equalsIgnoreCase("csv")) {
+            m_seperator = ',';
+        } else if (type.equalsIgnoreCase("tsv")) {
+            m_seperator = '\t';
+        } else {
+            throw new IllegalArgumentException("Error: type must be one of CSV or TSV");
+        }
+
+        validateStream();
+        // Kinesis firehose limits record size to 1,000 KB
+        setRowLengthLimit(1000000, m_skipInternals);
```

Comment:
No, according to the Amazon Kinesis Firehose doc the size limit of each record is 1,000KB